src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/SupportController.java

```diff
@@ -134,12 +145,27 @@ public void uploadDocumentToDocumentManagement(
             case SETTLEMENT_AGREEMENT:
                 documentsService.generateSettlementAgreement(claim.getExternalId(), authorisation);
                 break;
+            case DEFENDANT_PIN_LETTER:
+                documentsService.generateDefendantPinLetter(claim.getExternalId(), authorisation);
```

Comment:
Wouldn't it be better to always return the new document that has been generated or null if none was generated. That would give us a cleaner way to identify if a document has been created or not instead of `isDocumentCreated` method.

Reply:
that wouldn't work in case of other documents because even when an error occurs we still generate the document on the fly and return it. Only checking if the document exists in the claim is a sure shot way to determine if it was created successfully.